weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CompareEvaluationsPage/ecpTypes.ts

```diff
@@ -34,6 +35,9 @@ export type EvaluationComparisonSummary = {
   // SummaryMetrics define the metrics that are associated with the evaluation as a whole
   // often aggregated from the scoreMetrics.
   summaryMetrics: MetricDefinitionMap;
+
+  // Cache of the evaluation calls for Imperative Evaluations
+  _evaluationCallCache?: {[callId: string]: EvaluationEvaluateCallSchema};
```

Comment:
this seems unnecessary here as it is not intended to be (nor seems to actually be) consumed by any external code.

Reply:
Oh - wait, i see - you are passing this along over to the `fetchEvaluationComparisonResults`.

Reply:
it's used in 2 fetch funcs inside  `tsDataModelHooksEvaluationComparison.ts`

Reply:
but then all it is used for is to calculate `hasImperativeEvals`. I think what you might want to do instead is extend `EvaluationCall` to have an `isImperative` boolean flag which will make this more organized

Reply:
this seems to be the only usage
```
  const evaluationCallCache = summaryData._evaluationCallCache || {};
  const hasImperativeEvals = Object.keys(evaluationCallCache).some(
    id => id in evaluationCallCache && isImperative(evaluationCallCache[id])
  );
```